examples/experimental/dagster-dlift/dagster_dlift/project.py

```diff
@@ -41,3 +41,27 @@ def compute_data(self) -> "DbtCloudProjectEnvironmentData":
         return compute_environment_data(
             project_id=self.project_id, environment_id=self.environment_id, client=self.client
         )
+
+    def get_specs(
+        self, translator_cls: Type[DagsterDbtCloudTranslator]
+    ) -> Sequence[Union[AssetSpec, AssetCheckSpec]]:
+        from dagster_dlift.loader import DbtCloudProjectEnvironmentDefsLoader
+
+        data = DbtCloudProjectEnvironmentDefsLoader(project_environment=self).get_or_fetch_state()
+        translator = translator_cls(context=data)
+        all_external_data = [
+            *data.models_by_unique_id.values(),
+            *data.sources_by_unique_id.values(),
+            *data.tests_by_unique_id.values(),
+        ]
+        return [translator.get_spec(data) for data in all_external_data]
+
+    def get_asset_specs(
+        self, translator_cls: Type[DagsterDbtCloudTranslator] = DagsterDbtCloudTranslator
+    ) -> Sequence[AssetSpec]:
+        return [spec for spec in self.get_specs(translator_cls) if isinstance(spec, AssetSpec)]
```

Comment:
should we cache `get_specs`, or have `get_specs` call these instead? not a huge deal, but does mean we run `get_spec(data)` for all data entries 2x if a user calls `get_asset_specs()` once and `get_check_specs()` once

Reply:
Yea I buy that we should cache it.